esx_service/utils/vm_listener.py

```diff
@@ -212,9 +211,13 @@ def set_device_detached(vm_moref):
         # if it is a dvs managed volume, set its status as detached
         vmdk_path = vmdk_utils.find_dvs_volume(dev)
         if vmdk_path:
-            logging.info("Setting detach status for %s", vmdk_path)
+            readable_vmdk_path = vmdk_ops.gets_readable_path(vmdk_path)
```

Comment:
I think checking for conversion errors adds tons of code and loses the original value, without a real reason for it. 

It would be easier if gets_readable_path(vmdk_path) (btw, why `gets` and not `get`?) would be just returning the original path in case of an error. Then the lines 214->218 would be simply
```python
 logging.info("Setting detach status for %s", vmdk_ops.get_readable_path(vmdk_path))